Classes/Queue/DoctrineQueue.php

```diff
@@ -301,24 +376,16 @@ protected function resolveScheduledQueryPart(array $options): string
         if (!isset($options['delay'])) {
             return 'null';
         }
-        switch ($this->connection->getDatabasePlatform()->getName()) {
-            case 'sqlite':
-                return 'datetime(\'now\', \'+' . (int)$options['delay'] . ' second\')';
-            case 'postgresql':
-                return 'NOW() + INTERVAL \'' . (int)$options['delay'] . ' SECOND\'';
-            default:
-                return 'DATE_ADD(NOW(), INTERVAL ' . (int)$options['delay'] . ' SECOND)';
-        }
+
+        return $this->connection->getDatabasePlatform()->getDateAddSecondsExpression($this->connection->getDatabasePlatform()->getCurrentTimestampSQL(), (int)$options['delay']);
     }
 
     protected function getScheduledQueryConstraint(QueryBuilder $qb): CompositeExpression
     {
-        switch ($this->connection->getDatabasePlatform()->getName()) {
-            case 'sqlite':
-                return '(scheduled IS NULL OR scheduled <= datetime("now"))';
-            default:
-                return '(scheduled IS NULL OR scheduled <= NOW())';
-        }
+        return $qb->expr()->or(
```

Comment:
Maybe you didn't want to change too much, but this could as well mutate the query builder (like we do in many places for the Neos 9 projections, e.g. in the [ContentSubgraph](https://github.com/neos/neos-development-collection/blob/9.0/Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ContentSubgraph.php#L508))

```
protected function addScheduledQueryConstraints(QueryBuilder $qb): void
{
    $qb->andWhere(
        $qb->expre()->or(
            $qb->expr()->isNull('scheduled'),
            $qb->expr()->lte('scheduled', $this->connection->getDatabasePlatform()->getCurrentTimestampSQL())
        )
    );
}
```

but I'm not sure if that's any better really :)

Reply:
I'll leave that one as it is… 🤷